tests/test_resolver.py

```diff
@@ -1,18 +1,5 @@
-#
-#   Copyright 2023 The Foundry Visionmongers Ltd
-#
-#   Licensed under the Apache License, Version 2.0 (the "License");
-#   you may not use this file except in compliance with the License.
-#   You may obtain a copy of the License at
-#
-#       http://www.apache.org/licenses/LICENSE-2.0
-#
-#   Unless required by applicable law or agreed to in writing, software
-#   distributed under the License is distributed on an "AS IS" BASIS,
-#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-#   See the License for the specific language governing permissions and
-#   limitations under the License.
-#
+# SPDX-License-Identifier: Apache-2.0
```

Comment:
[question] What do you think about adding `black` and `pylint` for the python into CI as per other proj - it has been known to catch some useful things before now.

Reply:
The sentiment for the issue was "get all the boilerplate done so we can crack on with coding", happy to in another PR, but definitely before we start writing much more in the way of tests etc... to save churn.

Reply:
By this do you mean essentially mirroring [code_quality.yml](https://github.com/OpenAssetIO/OpenAssetIO/blob/f186bc529325e0f6858644848dd71054a732be56/.github/workflows/code-quality.yml) like we have in the main repo? Shouldn't be too much work if that's what you've in mind.

Reply:
Yeah that thing.

Reply:
This wasn't as simple as I thought, nonetheless i'm pretty nearing complete with the changes for it.

Thought that it might be best to deal with any subsequent conversation from the other addressed issues, then close this PR, before almost immediately opening another with the changes for this. It's a fair bit of new stuff that's not really related to the thrust of this change, and I think would be better in another review.

Will talk about this in standup anyhow.

Reply:
Decided to extract this change from this PR, do it separately.